Implement FCFS scheduling algorithm structure

kirti734 opened the pull request:
This file implements a First-Come, First-Served (FCFS) scheduling algorithm with necessary data structures and function prototypes.

kirti734 committed:
Implement FCFS scheduling algorithm structure

This file implements a First-Come, First-Served (FCFS) scheduling algorithm with necessary data structures and function prototypes.

kirti734 committed:
Update FCFS_Implementation.c

kirti734 committed:
Introduce sleep for 1 second in process handling

Added a sleep function to pause execution for 1 second.

priyankagupta-ITT commented on FCFS_Implementation.c:
```diff
@@ -0,0 +1,693 @@
+		
+		handleIOprocess();
+
+		sleep(1);
```
What do you need to put sleep(1)??

kirti734 replied:
Because the PDF explicitly says  ‘Simulate CPU burst using: sleep(1) per time unit’.
 so I assumed I should use it.
 
 But Now I remove the sleep(1)

priyankagupta-ITT commented on FCFS_Implementation.c:
```diff
@@ -0,0 +1,693 @@
+            }
+            else if (runningProcess != NULL && runningProcess -> isKilled)
+            {
+                runningProcess = NULL;
```
if running process is killed then you should add this process in terminated queue.

kirti734 replied:
I have already enqueue the killed process in the terminated queue in the processKillEventForCurrentTime() function

priyankagupta-ITT commented on FCFS_Implementation.c:
```diff
@@ -0,0 +1,693 @@
+	initializeQueue(&waitingQueue);
+	initializeQueue(&terminatedQueue);
+	
+	pcbTable = (struct PCB**) malloc (HASHMAP_SIZE * sizeof(struct PCB*));
```
You are defining PCB table here. Better to use calloc instead of malloc to initialize the memory..

kirti734 replied:
Now,I have used the calloc instead of malloc for the initialization of the memory

kirti734 committed:
Refactor memory allocation and process state handling

Replaced malloc with calloc for pcbTable allocation and improved process state management.
